Suppress events about internal audit nodes

escowles opened the pull request:
Fixes https://jira.duraspace.org/browse/FCREPO-1506

acoburn commented on audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/EventRouter.java:
```diff
@@ -43,7 +44,9 @@ public void configure() throws Exception {
-            .to("direct:event");
+            .choice()
+            .when(not(header("org.fcrepo.jms.identifier").startsWith(simple("{{audit.container}}"))))
+                .to("direct:event");
```
this works, but filter(...) would be simpler

    from("...")
        .routeId("AuditFcrepoRouter")
        .filter(not(header(JmsHeaders.IDENTIFIER).startsWith(simple("..."))))
            .to("direct:event")

Also note that the fcrep-camel component has CONST values for the JMS headers

acoburn commented on audit-triplestore-web/src/main/resources/application.properties:
```diff
@@ -3,3 +3,4 @@ jms.fcrepoEndpoint=${fcrepo.jms.endpoint:topic:fedora}
 jms.brokerUrl=failover:(tcp://${fcrepo.jms.host:localhost}:${fcrepo.dynamic.jms.port:61616})?startupMaxReconnectAttempts=1&initialReconnectDelay=1
 triplestore.baseUrl=${fcrepo.audit.triplestore.baseUrl:localhost:3030/test/update}
 event.baseUri=${fcrepo.audit.baseUri:http://example.com/event}
+audit.container=${fcrepo.audit.container:/audit}
```
should the default value have a trailing slash? in case there is e.g. an `/auditory` node in the repository

escowles replied:
I dithered about this -- but half of the events are about /audit itself.  I'll try doing a check for startsWith(/audit/) or equals(/audit).

acoburn replied:
good idea. The PredicateBuilder class has an `or` function for this very thing:

    filter(not(or(<Expression1>, <Expression2>)))
      .to("destination")

acoburn commented on audit-triplestore/src/main/resources/OSGI-INF/blueprint/blueprint.xml:
```diff
@@ -16,6 +16,7 @@
        <cm:property name="triplestore.baseUrl" value="localhost:3030/test/update"/>
        <!-- Base URI to be used in contructing the URI for the JMS event -->
        <cm:property name="event.baseUri" value="http://example.com/event"/>
+       <cm:property name="audit.container" value="/audit"/>
```
see above about a trailing slash

acoburn commented:
Some updates to the two README files about the new configuration option would be nice

escowles commented:
I've updated EventRouter to use a filter() and handle other containers that start with /audit (e.g., /auditions/), and updated the tests and readme.

acoburn commented:
Resolved via 5c7912a47101f95ca4aac7cb3ef5f0edf7dfc912
